js/filter.js

```diff
@@ -7,20 +7,63 @@
   var removePins = window.pin.remove;
 
   var mapFiltersFormChangeHandler = function () {
-    updatePins();
+    window.debounce.call(updatePins());
```

Comment:
Тут действительно неправильно используется debounce, он должен принимать функцию и он возвращает функцию, а у тебя в аргументах функция сразу вызывается. То есть, вот так будет работать
```suggestion
    window.debounce.call(updatePins)();
```